keras/utils/timeseries_dataset.py

```diff
@@ -209,7 +209,7 @@ def timeseries_dataset_from_array(
 
     # Determine the lowest dtype to store start positions (to lower memory
     # usage).
-    num_seqs = end_index - start_index - (sequence_length * sampling_rate) + 1
+    num_seqs = end_index - start_index - ((sequence_length - 1) * sampling_rate)
```

Comment:
the extra bracket for `(sequence_length - 1) * sampling_rate` can be skipped.

Reply:
Okay, removed them in the new commit